src/Compilers/CSharp/Test/Emit3/PartialEventsAndConstructorsTests.cs

```diff
@@ -1131,6 +1133,188 @@ static void verifyMetadata(ModuleSymbol module)
         }
     }
 
+    [Fact]
+    public void GetDeclaredSymbol()
+    {
+        var source = ("""
+            partial class C
+            {
+                public partial event System.Action E, F;
+                public partial event System.Action E { add { } remove { } }
+                public partial event System.Action F { add { } remove { } }
+
+                public partial C();
+                public partial C() { }
+            }
+            """, "Program.cs");
+
+        var comp = CreateCompilation(source).VerifyDiagnostics();
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+
+        var eventDefs = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().ToImmutableArray();
+        Assert.Equal(2, eventDefs.Length);
+
+        var eventImpls = tree.GetRoot().DescendantNodes().OfType<EventDeclarationSyntax>().ToImmutableArray();
+        Assert.Equal(2, eventImpls.Length);
+
+        {
+            var defSymbol = (IEventSymbol)model.GetDeclaredSymbol(eventDefs[0])!;
+            Assert.Equal("event System.Action C.E", defSymbol.ToTestDisplayString());
+
+            IEventSymbol implSymbol = model.GetDeclaredSymbol(eventImpls[0])!;
+            Assert.Equal("event System.Action C.E", implSymbol.ToTestDisplayString());
+
+            Assert.NotEqual(defSymbol, implSymbol);
+            Assert.Same(implSymbol, defSymbol.PartialImplementationPart);
+            Assert.Same(defSymbol, implSymbol.PartialDefinitionPart);
```

Comment:
nit: seems like it would be good to assert that the `implSymbol.PartialImplementationPart` is null, same with `defSymbol.PartialDefinitionPart`.